model/types.go

```diff
@@ -64,3 +64,43 @@ type State interface {
 	GetDataOutputSchema() string
 	GetMetadata() Metadata_1
 }
+
+// Permits transitions to other states based on data conditions
+type Databasedswitch struct {
+	// Defines conditions evaluated against state data
+	DataConditions        []DatabasedswitchDataConditionsElem `json:"dataConditions,omitempty"`
+	DatabasedswitchCommon DatabasedswitchCommon
+}
+
+type DatabasedswitchCommon struct {
+	// URI to JSON Schema that state data input adheres to
+	DataInputSchema *string `json:"dataInputSchema,omitempty"`
+
+	// URI to JSON Schema that state data output adheres to
+	DataOutputSchema *string `json:"dataOutputSchema,omitempty"`
+
+	// Default transition of the workflow if there is no matching data conditions. Can
+	// include a transition or end definition
+	Default *Defaultdef `json:"default,omitempty"`
+
+	// Unique State id
+	Id *string `json:"id,omitempty"`
+
+	// Metadata corresponds to the JSON schema field "metadata".
+	Metadata Metadata_1 `json:"metadata,omitempty"`
+
+	// State name
+	Name *string `json:"name,omitempty"`
+
+	// States error handling and retries definitions
+	OnErrors []Error `json:"onErrors,omitempty"`
+
+	// State start definition
+	Start *Start `json:"start,omitempty"`
+
+	// StateDataFilter corresponds to the JSON schema field "stateDataFilter".
+	StateDataFilter *Statedatafilter `json:"stateDataFilter,omitempty"`
+
+	// State type
+	Type *string `json:"type,omitempty"`
```

Comment:
These two structures are a copy and paste from the generated code? I wouldn't like to take this approach since we would have to start manually updating this file whenever we change the spec. Is it possible to make this change directly in the tooling?

Reply:
Yes they are a simple copy-paste, just to include most of them in a `common` struct, following the `workflow` type paradigm